### DISCUSSION TITLE
This set_column map code doesn't make sense

### DISCUSSION THREAD

**Discussion by mbeynon** [datanymizer/datanymizer#162] [General]
Here the get_columns() call uses the TABLE_COLUMN_QUERY with "ORDER BY cc.ordinal_position ASC":

https://github.com/datanymizer/datanymizer/blob/main/datanymizer_dumper/src/postgres/schema_inspector.rs#L147

When `set_columns()` is called, the code below runs.  The `columns` arg is in position order already.  This code then creates `column_refs`, sorts it by the `.position` value, and creates a map from string name to the index of column_refs.  Then it saves the columns vec and the map, but lets column_refs go out of scope.

https://github.com/datanymizer/datanymizer/blob/2dc82b3eb0022cd359d391ead9232a3791d300e9/datanymizer_dumper/src/postgres/table.rs#L94-L104

I don't see the point of column_refs since it's the same order as columns is to start, and if it was different somehow, the map would be invalid because it doesn't point to column indexes.  Fix appears to be removing `column_refs` altogether, and create the map using `column` indexes.

Am I misunderstanding this?

**Comment by evgeniy-r** [comment]
Hello!

It seems you are right, we should remove sorting (and indicate this requirement for the argument somehow) or sort `columns` too.
